### ISSUE TITLE
Ansible operator should log playbook tasks to ansible log container as they're executed

### ISSUE THREAD

**Issue by chancez** [operator-framework/operator-sdk#1453]
## Feature Request

**Is your feature request related to a problem? Please describe.**

My playbook takes a long time to execute, and while I can see the high level overview of what's running within the operator container through the `[playbook task]` event logging, I cannot see the full output of each task until the entire playbook finishes executing, either successfully or through a failure.

The event logs from the operator container look like this, and only provide high level information from the task names.

```
{"level":"info","ts":1558143115.2085207,"logger":"logging_event_handler","msg":"[playbook task]","name":"operator-metering","namespace":"metering-chancez2","gvk":"metering.openshift.io/v1alpha1, Kind=MeteringConfig","event_type":"playbook_on_task_start","job":"6159221595405140213","EventData.Name":"meteringconfig : Delete secret resources with label selector metering.openshift.io/prune=presto-catalog-config-secret"}
{"level":"info","ts":1558143115.4031546,"logger":"logging_event_handler","msg":"[playbook task]","name":"operator-metering","namespace":"metering-chancez2","gvk":"metering.openshift.io/v1alpha1, Kind=MeteringConfig","event_type":"playbook_on_task_start","job":"6159221595405140213","EventData.Name":"meteringconfig : Helm template templates/presto/presto-common-config.yaml"}
{"level":"info","ts":1558143117.486677,"logger":"logging_event_handler","msg":"[playbook task]","name":"operator-metering","namespace":"metering-chancez2","gvk":"metering.openshift.io/v1alpha1, Kind=MeteringConfig","event_type":"playbook_on_task_start","job":"6159221595405140213","EventData.Name":"meteringconfig : Fail when helm template errors on templates/presto/presto-common-config.yaml"}
{"level":"info","ts":1558143117.6055598,"logger":"logging_event_handler","msg":"[playbook task]","name":"operator-metering","namespace":"metering-chancez2","gvk":"metering.openshift.io/v1alpha1, Kind=MeteringConfig","event_type":"playbook_on_task_start","job":"6159221595405140213","EventData.Name":"meteringconfig : Install templates/presto/presto-common-config.yaml"}
{"level":"info","ts":1558143120.2956345,"logger":"logging_event_handler","msg":"[playbook task]","name":"operator-metering","namespace":"metering-chancez2","gvk":"metering.openshift.io/v1alpha1, Kind=MeteringConfig","event_type":"playbook_on_task_start","job":"6159221595405140213","EventData.Name":"meteringconfig : Query for configmap with selector metering.openshift.io/prune=presto-common-config resources to delete"}
{"level":"info","ts":1558143120.5046253,"logger":"logging_event_handler","msg":"[playbook task]","name":"operator-metering","namespace":"metering-chancez2","gvk":"metering.openshift.io/v1alpha1, Kind=MeteringConfig","event_type":"playbook_on_task_start","job":"6159221595405140213","EventData.Name":"meteringconfig : Delete configmap resources with label selector metering.openshift.io/prune=presto-common-config"}
```

Additionally, this doesn't contain information like whether a task was skipped, so those logs are actually somewhat misleading, depending on how you've named your tasks.

But while this is happening, the `ansible` container has no logs until the playbook finishes executing. So for the first playbook run, if you look at the logs in there, it's just:

```
Setting up watches.  Beware: since -r was given, this may take a while!
Watches established.
```

**Describe the solution you'd like**

Any solution that streams the playbook logs as the playbook executes rather than at the end would be acceptable.

**Comment by chancez** [comment]
After some digging, I found how the logs work https://github.com/operator-framework/operator-sdk/blob/master/internal/pkg/scaffold/ansible/ao_logs.go#L40-L49

It seems it waits for the close_write event and uses `cat` which means it's waiting for the file to be closed, then using `cat` to write the entire contents out.

I'm currently working on an alternative script that streams these logs out, but it's definitely tricky. So far my current idea is to wait for `create` events, and use `(--tail --follow=${dir}/${file} || true) &` so that it tails the file in the background. Eventually the file gets removed, and the tail should exit when the file is deleted. 

This approach means `tail` running unnecessarily after the logs stop, but I'm having trouble figuring a good method of stopping it when the file is closed. Probably having two background jobs running inotify, one watching for create, one for closed, and starting/killing tail processes. The only issue is if you do this, you may end up killing tail before it logs everything out, since the file may be closed before tail finishes echoing the recent log output.

I'll keep experimenting, but I thought I'd mention my current thoughts here.

**Comment by fabianvf** [comment]
Yeah, the main reason I went with the `cat` on close approach is that if you have multiple CRs all reconciling at once the logs would interleave and it would make parsing the logs a little rough (might not be relevant right now, I'm not sure how many simultaneous jobs we run). Maybe we could tail on create and cat on close or something, though that might be a little janky.

**Comment by chancez** [comment]
Yah, it's easy enough to work-around by running my own sidecar logging container. I agree with interweaved logs, and it's possible to add more than 1 sidecar container if you want to have separate log streams.

**Comment by geerlingguy** [comment]
One other suggestion: could we use the yaml callback plugin for a little nicer output? (see: https://www.jeffgeerling.com/blog/2018/use-ansibles-yaml-callback-plugin-better-cli-experience)

**Comment by camilamacedo86** [comment]
Hi @chancez, 

We add an improvement regards the logs. See: https://github.com/operator-framework/operator-sdk/pull/2321. 

Could you please let us know if this improvement could not attend your scenario as well and if we can close this one?

**Comment by camilamacedo86** [comment]
Hi @chancez, 

All indicate that this scenario is solved in the PR: #2321.  It is merged in the master and will probably be available in the next release. So, we are closing this one. However, please feel free to re-open if you think that something still required to get addressed here.
